### PR TITLE
Fix #44: Use PSR-20 ClockInterface instead of TimerInterface

### LINKED ISSUE #44: PSR ClockInterface instead of TimerInterface
Should we replace `\Yiisoft\Yii\RateLimiter\Time\TimerInterface` with the one from https://www.php-fig.org/psr/psr-20?

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | #44

### CODE REVIEW

File: src/Counter.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public function hit(string $id): CounterState
         do {
             // Last increment time.
             // In GCRA it's known as arrival time.
-            $lastIncrementTimeInMilliseconds = $this->timer->nowInMilliseconds();
+            $lastIncrementTimeInMilliseconds = round((float)$this->timer->now()->format('U.u') * 1000);
```
Comment (vjik) [inline src/Counter.php:78]:
```suggestion
            $lastIncrementTimeInMilliseconds = round($this->timer->now()->format('U.u') * 1000);
```

File: tests/MiddlewareTest.php [hunk 2]
```diff
@@ -227,7 +227,7 @@ public function testWithLimitingFunction(): void
      */
     public function testWithExceedingMaxAttempts(): void
     {
-        $timer = new FrozenTimeTimer();
+        $timer = new FrozenClock();
```
Comment (vjik) [inline tests/MiddlewareTest.php:230]: Here we can don't use specific timer and replace it to `null`.

File: tests/MiddlewareTest.php [hunk 3]
```diff
@@ -259,7 +259,7 @@ public function testWithExceedingMaxAttempts(): void
 
     public function testFailStoreUpdatedDataMiddleware(): void
     {
-        $timer = new FrozenTimeTimer();
+        $timer = new FrozenClock();
```
Comment (vjik) [inline tests/MiddlewareTest.php:262]: Here we can don't use specific timer and replace it to `null`.

File: tests/Storage/StorageTest.php [hunk 4]
```diff
@@ -34,20 +34,20 @@ public function testGetKeyWithExistsKey(): void
     {
         $storage = $this->getStorage();
 
-        $want = (new FrozenTimeTimer())->nowInMilliseconds();
+        $want = round((float)(new DateTimeImmutable())->format('U.u') * 1000);
```
Comment (vjik) [inline tests/Storage/StorageTest.php:37]:
```suggestion
        $want = round((new DateTimeImmutable())->format('U.u') * 1000);
```

File: tests/Storage/StorageTest.php [hunk 5]
```diff
@@ -34,20 +34,20 @@ public function testGetKeyWithExistsKey(): void
...
-        $this->assertEquals($result, $want);
+        $this->assertEquals($want, $result);
     }
 
     public function testSaveIfNotExistsWithNewKey(): void
     {
         $storage = $this->getStorage();
 
-        $value = (new FrozenTimeTimer())->nowInMilliseconds();
+        $value = round((float)(new DateTimeImmutable())->format('U.u') * 1000);
```
Comment (vjik) [inline tests/Storage/StorageTest.php:50]:
```suggestion
        $value = round((new DateTimeImmutable())->format('U.u') * 1000);
```

File: tests/Storage/StorageTest.php [hunk 6]
```diff
@@ -58,7 +58,7 @@ public function testSaveCompareAndSwapWithExistsKeyAndOldValueSame(): void
     {
         $storage = $this->getStorage();
 
-        $oldValue = (new FrozenTimeTimer())->nowInMilliseconds();
+        $oldValue = round((float)(new DateTimeImmutable())->format('U.u') * 1000);
```
Comment (vjik) [inline tests/Storage/StorageTest.php:61]:
```suggestion
        $oldValue = round((new DateTimeImmutable())->format('U.u') * 1000);
```

File: tests/Storage/ApcuStorageTest.php [hunk 7]
```diff
@@ -60,11 +60,11 @@ public function testSaveCompareAndSwapWithNewKey(): void
         $this->assertFalse($result);
     }
 
-    public function testSaveCompareAndSwapWithExistsKeyButOldValueDiffrent(): void
+    public function testSaveCompareAndSwapWithExistsKeyButOldValueDifferent(): void
     {
         $storage = $this->getStorage();
 
-        $oldValue = (new FrozenTimeTimer())->nowInMilliseconds();
+        $oldValue = round((float)(new DateTimeImmutable())->format('U.u') * 1000);
```
Comment (vjik) [inline tests/Storage/ApcuStorageTest.php:67]:
```suggestion
        $oldValue = round((new DateTimeImmutable())->format('U.u') * 1000);
```

File: tests/Storage/ApcuStorageTest.php [hunk 8]
```diff
@@ -47,7 +47,7 @@ public function testSaveCompareAndSwapWithNewKey(): void
     {
         $storage = $this->getStorage();
 
-        $newValue = (new FrozenTimeTimer())->nowInMilliseconds();
+        $newValue = round((float)(new DateTimeImmutable())->format('U.u') * 1000);
```
Comment (vjik) [inline tests/Storage/ApcuStorageTest.php:50]:
```suggestion
        $newValue = round((new DateTimeImmutable())->format('U.u') * 1000);
```

File: tests/Storage/ApcuStorageTest.php [hunk 9]
```diff
@@ -35,7 +35,7 @@ public function testSaveIfNotExistsWithExistsKey(): void
     {
         $storage = $this->getStorage();
 
-        $value = (new FrozenTimeTimer())->nowInMilliseconds();
+        $value = round((float)(new DateTimeImmutable())->format('U.u') * 1000);
```
Comment (vjik) [inline tests/Storage/ApcuStorageTest.php:38]:
```suggestion
        $value = round((new DateTimeImmutable())->format('U.u') * 1000);
```

File: tests/CounterTest.php [hunk 10]
```diff
@@ -20,11 +20,11 @@ protected function getStorage(): StorageInterface
 
     /**
      * Testing that in concurrent scenarios, when dirty reads occur,
-     * the current limiter cannot be as expected By 'SimpleCacheStorage'.
+     * the current limiter can't be as expected By 'SimpleCacheStorage'.
      */
     public function testConcurrentHitsWithDirtyReading(): void
     {
-        $timer = new FrozenTimeTimer();
+        $timer = new FrozenClock();
```
Comment (vjik) [inline tests/CounterTest.php:27]: Here we can don't use specific timer and replace it to `null`.

File: src/Counter.php (hunk 1 above)
Comment (samdark) [reply src/Counter.php:78]: That would make static analysis complain.

**Pull request closed without merging**